lib/oneview-sdk/resource/api500/c7000/storage_pool.rb

```diff
@@ -96,6 +96,13 @@ def request_refresh
           update
           refresh
         end
+
+        # Sets the storage system
+        # @param [OneviewSDK::StorageSystem] storage_system
+        def set_storage_system(storage_system)
+          raise IncompleteResource, 'Please set the storage system\'s uri attribute!' unless storage_system['uri']
```

Comment:
I'd rather to use retrieve! Because I can pass the storage system instance just with the name.

Reply:
yep, but I just did ctrl+c ctrl+v of similar code of API200. haha
Maybe this solution is simplist, but is just a helper method, the user can apply the retrieve! before this method too.